### PR TITLE
feat: Add config option for providing AWS account ID for linking

### PR DESCRIPTION
**Pull request by chynesNR**
Adds the ability to provide an AWS Account ID, which can be used to help link related entities.

This functionality is needed for multiple feature branches which is why I'm merging to `main` first, even though this won't be used yet.

### CODE REVIEW

**Review by nrcventura — COMMENTED** [review]
Looks good to me. I think we just need a few more test updates to increase the patch coverage.

I'm also assuming that the updates to the instrumentation that will use this setting will be handled in separate PRs.

**Review by tippmar-nr — APPROVED** [review]

**Comment by chynesNR** [comment]
> Looks good to me. I think we just need a few more test updates to increase the patch coverage.
> 
> I'm also assuming that the updates to the instrumentation that will use this setting will be handled in separate PRs.

Correct, there are separate tasks for that work and testing

**Review by nrcventura — APPROVED** [review]

**Review by tippmar-nr — APPROVED** [review]

File: tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs [hunk 1]
```diff
@@ -4439,8 +4439,93 @@ public void AzureFunctionServiceName_ShouldReturnServiceName_WhenEnvironmentVari
             Assert.That(result, Is.EqualTo("some-service-name"));
         }
 
+        #endregion Cloud
```
Comment (nr-ahemsath) [inline tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs:4442]: Is this where the `Cloud` region should end?

Comment (chynesNR) [reply tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs:4442]: Well that's weird. Not sure how I swapped the begin and end. Good catch!

**Review by nrcventura — APPROVED** [review]

**Pull request merged**